packages/api-explorer/src/scenes/utils/hooks/tagStoreSync.spec.ts

```diff
@@ -0,0 +1,102 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2022 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import { renderHook } from '@testing-library/react-hooks'
+import { useHistory } from 'react-router-dom'
+import type { Location } from 'history'
+import * as reactRedux from 'react-redux'
+import * as routerLocation from 'react-router-dom'
+import { createTestStore, withReduxProvider } from '../../../test-utils'
+import { useTagStoreSync } from './tagStoreSync'
+
+jest.mock('react-router', () => {
+  const ReactRouter = jest.requireActual('react-router')
+  return {
+    ...ReactRouter,
+    useHistory: jest.fn().mockReturnValue({ push: jest.fn(), location }),
+    useLocation: jest.fn().mockReturnValue({ pathname: '/', search: '' }),
+  }
+})
+
+describe('useTagStoreSync', () => {
+  const mockDispatch = jest.fn()
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('does nothing if uninitialized', () => {
+    const { push } = useHistory()
+    const wrapper = ({ children }: any) => withReduxProvider(children)
+    renderHook(() => useTagStoreSync(), { wrapper })
+    expect(push).not.toHaveBeenCalled()
+  })
+
+  describe.each([
```

Comment:
test.each?

Reply:
I would be nesting tests if I replaced this with test.each. The intention was to account for the 2 different tabs, methods and types, where this filtering logic must correctly apply. Each tab has their own respective filters, and I was initially considering the case where we are in a tab (say methods) with the URL containing a type tag filter (say v=specification). I decided against having that explicit of a test though because I believe this is covered by my testing of "invalid" filters.

Thoughts?

Reply:
I misread, this is fine as is.